packages/api-client/src/tcp/WebSocketClient.ts

```diff
@@ -138,25 +135,19 @@ export class WebSocketClient extends EventEmitter {
    * Essentially the websocket will lock before execution of this function and
    * unlocks after the execution of the handler and pushes all buffered messages.
    */
-  public async connect(clientId?: string, onConnect?: OnConnect): Promise<WebSocketClient> {
-    if (onConnect) {
-      this.onConnect = async () => {
-        this.abortHandler = new AbortHandler();
-        try {
-          this.logger.info('Calling "onConnect"');
-          await onConnect(this.abortHandler);
-        } catch (error) {
-          this.logger.warn(`Error during execution of "onConnect"`, error);
-          this.emit(WebSocketClient.TOPIC.ON_ERROR, error);
-        }
-        this.onStateChange(this.socket.getState());
-      };
```

Comment:
The problem lies here in the fact that we would call the onConnect before actually being connected to the websocket.
Which means that we start processing the notification stream but there could be messages that come in between the moment we start processing the notification stream and the moment the websocket is actually connected (and thus we have potential missing messages)